server/src/main/java/org/elasticsearch/common/settings/AbstractScopedSettings.java

```diff
@@ -296,6 +296,55 @@ public void apply(Map<String, Tuple<A, B>> values, Settings current, Settings pr
         });
     }
 
+    /**
+     * Adds a affix settings consumer that accepts the settings for a group of settings. The consumer is only
+     * notified if at least one of the settings change.
+     * <p>
+     * Note: Only settings registered in {@link SettingsModule} can be changed dynamically.
+     * </p>
+     */
+    public synchronized void addAffixGroupUpdateConsumer(List<Setting.AffixSetting<?>> settings, BiConsumer<String, Settings> consumer) {
```

Comment:
Can you pull this out into a separate change, leaving the usage here for reference? Then we can integrate this back after that has a separate review from the core/infra team.

Reply:
I opened #37679